internal/runners/activate/activate.go

```diff
@@ -97,25 +96,27 @@ func (r *Activate) run(params *ActivateParams, activatorLoop activationLoopFunc)
 func (r *Activate) setupPath(namespace string, preferredPath string) (string, error) {
 	var (
 		targetPath string
-		err        error
+		proj       *project.Project
+		fail       *failures.Failure
 	)
 
 	switch {
 	// Checkout via namespace (eg. state activate org/project) and set resulting path
 	case namespace != "":
-		targetPath, err = r.namespaceSelect.Run(namespace, preferredPath)
+		namesPath, err := r.namespaceSelect.Run(namespace, preferredPath)
+		if err != nil {
+			return "", err
+		}
+		proj, fail = project.FromPath(namesPath)
+		targetPath = namesPath
 	// Use the user provided path
 	case preferredPath != "":
-		targetPath, err = preferredPath, nil
+		proj, fail = project.FromPath(preferredPath)
+		targetPath = preferredPath
 	// Get path from working directory
 	default:
-		targetPath, err = osutils.Getwd()
+		proj, fail = project.GetSafe()
 	}
-	if err != nil {
-		return "", err
-	}
-
-	proj, fail := project.FromPath(targetPath)
 	if fail != nil {
 		return targetPath, fail
```

Comment:
I think this line causes some errors to be missed. The correct behavior seems to be occurring, so I didn't attempt any debugging. If `return targetPath, fail` is exchanged with `return targetPath, fail.ToError()`, the outcome was different. This might be due to me misunderstanding the ToError/fail/error interactions.

Reply:
Please feel free to change it to `ToError()`, you're right this looks bugged.